optimizer/optimizer.nim

```diff
@@ -0,0 +1,169 @@
+#
+#
+#             C Optimizer
+#        (c) Copyright 2020 Andreas Rumpf
+#
+#    See the file "copying.txt", included in this
+#    distribution, for details about the copyright.
+#
+
+## An optimizer that optimizes the C code that Nim produces
```

Comment:
why not fix the code generation instead of generating a c file, parsing it, and post-processing it, which seems more fragile, roundabout and less efficient?